backend/app/user/routes.py

```
@@ -55,3 +66,62 @@
+    background_tasks: BackgroundTasks = BackgroundTasks(),
+    current_user: dict = Depends(get_current_user),
+):
+    user_id = str(current_user["_id"])
+
+    try:
+        # Validate, process, upload
+        urls = await storage_service.upload_image_workflow(
+            file=file, folder="users", entity_id=user_id
+        )
+    except ValueError as ve:
+        raise HTTPException(status_code=400, detail=str(ve))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)